src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -435,23 +435,35 @@ private void walk(TypedVar symbol, boolean isDefault) {
         // Closure represents top-level functions as classes because they might be new-able.
         // This happens through externs es3.js which has Function marked as constructor.
         // See https://github.com/angular/closure-to-dts/issues/90
-        boolean ordinaryFunctionAppearingAsClass = type.isConstructor() &&
-            type.getDisplayName().equals("Function");
+        boolean ordinaryFunctionAppearingAsClass =
+            ftype.isConstructor() && "Function".equals(ftype.getDisplayName());
 
-        if (type.isOrdinaryFunction() || ordinaryFunctionAppearingAsClass) {
+        if (ftype.isOrdinaryFunction() || ordinaryFunctionAppearingAsClass) {
           emit("function");
           emit(getUnqualifiedName(symbol));
           visitFunctionDeclaration(ftype);
           emit(";");
           emitBreak();
           return;
+        } else if (!ftype.isNominalConstructor()) {
+          // A top-level field that has a specific constructor function type.
```

Comment:
this is identical with code at line 515, but it might not be worth refactoring to reuse it. At least reuse `visitTypeDeclaration` instead of `visitType`